### PR TITLE
Add iOS-only systemItem prop for buttons in Navigator.Config

### PR DESCRIPTION
**Pull request by nfcampos**
Hello :)

This PR introduces a `systemItem` property for `Button` in `Navigator.Config` to allow usage of `UIBarButtonSystemItem` for buttons in the navigation bar (iOS-only, not sure if there is an equivalent in android). 
This provides easy access to some very common icons and localised buttons on iOS, such as add, edit, done, search, etc, (full list available here https://developer.apple.com/reference/uikit/uibarbuttonsystemitem)
The only one I didn't implement `fixedSpace` as it requires a `width` parameter that I'm not sure how to expose.

## Changes
- Implement `systemItem` property on `Button`
- Updated the docs
- Added a new screen to the example app showing some of the ways the NavigationBar can be configured (I'll see if I find some time to update this screen with even more entries in another PR)

An example with a right button using `add`:
screen shot 2017-05-03 at 22 50 29

### CODE REVIEW

File: lib/ios/native-navigation/ReactNavigationImplementation.swift [hunk 1]
```diff
@@ -192,6 +200,36 @@ func barButtonStyleFromString(_ string: String?) -> UIBarButtonItemStyle {
   }
 }
 
+func barButtonSystemItemFromString(_ string: String?) -> UIBarButtonSystemItem? {
+  switch string {
+  case .some("done"): return .done
```
Comment (benkraus) [inline lib/ios/native-navigation/ReactNavigationImplementation.swift]: Can't we just check like: `case "done":`? Might take out the verbosity here

**Comment by benkraus** [comment]
Dang, I just merged something else. :| Can you rebase so we can get this merged?

File: lib/ios/native-navigation/ReactNavigationImplementation.swift [hunk 2]
```diff
@@ -51,16 +51,24 @@ class BlockBarButtonItem: UIBarButtonItem {
...
     image: UIImage?,
+    barButtonSystemItem: UIBarButtonSystemItem?,
     style: UIBarButtonItemStyle,
     enabled: Bool?,
     tintColor: UIColor?,
     titleTextAttributes: [String: Any]?
   ) {
     if let title = title {
       self.init(title: title, style: style)
+    } else if let barButtonSystemItem = barButtonSystemItem {
```
Comment (benkraus) [inline lib/ios/native-navigation/ReactNavigationImplementation.swift]: Just a side note - in iOS, it's pretty much the fact that you can't have a mixed bar button type of a system vs one with a title/image. In the future, I'd love to support setting both a title AND an image which can be done through the `customView` initializer on `UIBarButtonItem`. That being said, it'd make sense to have this as the very first case, then if it isn't a system item, go on to title/image etc. Not a big deal now, yet. We can either bump this up now, or perhaps do it in the future, no big deal either way.

**Comment by henrikra** [comment]
@nfcampos Are you still on this?

File: lib/ios/native-navigation/ReactNavigationImplementation.swift (hunk 2 above)
Comment (nfcampos) [inline lib/ios/native-navigation/ReactNavigationImplementation.swift]: moved up

**Comment by nfcampos** [comment]
@benkraus @henrikra totally forgot about this, i've rebased now

File: lib/ios/native-navigation/ReactNavigationImplementation.swift [hunk 3]
```diff
@@ -51,15 +51,23 @@ class BlockBarButtonItem: UIBarButtonItem {
...
     image: UIImage?,
+    barButtonSystemItem: UIBarButtonSystemItem?,
     style: UIBarButtonItemStyle,
     enabled: Bool?,
     tintColor: UIColor?,
     titleTextAttributes: [String: Any]?
   ) {
-    if let title = title {
+    if let barButtonSystemItem = barButtonSystemItem {
+      self.init(barButtonSystemItem: barButtonSystemItem)
```
Comment (benkraus) [inline lib/ios/native-navigation/ReactNavigationImplementation.swift]: Do we need this new convenience initializer? Can we just call `super.init(barButtonSystemItem: barButtonSystemItem, target: self, action: #selector(barButtonItemPressed))`?

Comment (nfcampos) [inline lib/ios/native-navigation/ReactNavigationImplementation.swift]: I was just following the style of the other convenience initialisers. I personally don't see the value, but I didn't want to change it in this PR

Comment (benkraus) [inline lib/ios/native-navigation/ReactNavigationImplementation.swift]: Ok, it's really just cleaner is all. Not a big deal. I say we merge!

**Comment by benkraus** [comment]
We should get this project set up on a CI system of some sorts...

**Comment by nfcampos** [comment]
yeah definitely

**Pull request merged**